### PR TITLE
replaced grains tests with canonical-data tests and fixed example script

### PR DESCRIPTION
**Pull request by dhpollack**
fix for bug related to #140

### CODE REVIEW

File: exercises/grains/runtests.jl [hunk 1]
```diff
@@ -3,10 +3,14 @@ using Test
 include("grains.jl")
```
Comment (SaschaMann) [inline exercises/grains/runtests.jl]:
```suggestion
# canonical data version: 1.2.0

include("grains.jl")
```

This isn't documented and not consistent among all test suites yet, but adding a version comment makes updating easier later on.

File: exercises/grains/runtests.jl [hunk 2]
```diff
@@ -3,10 +3,14 @@ using Test
...
-        @test on_square(s) == 2^(s-1)
-        @test total_after(s) == 2^s - 1
-    end
+    @test on_square(1) == 1
+    @test on_square(2) == 2
+    @test on_square(3) == 4
+    @test on_square(4) == 8
+    @test on_square(16) == 32768
+    @test on_square(32) == 2147483648
+    @test on_square(64) == 9223372036854775808
```
Comment (SaschaMann) [inline exercises/grains/runtests.jl]: Perhaps for the sake of teaching the cases that require an `Int128` should be in their own testset, so that there's some "progression": first one has to solve the algorithm part, then making sure it works for the entire domain.

 I'll leave this decision up to you, though, I'm fine with it as it is, too.

Comment (dhpollack) [reply exercises/grains/runtests.jl]: that's actually a good idea.  Do you mind if I change the names of the testsets?  Perhaps calling the first set, "beginning squares" and then "ending squares" for the later tests?

Comment (dhpollack) [reply exercises/grains/runtests.jl]: also, if I add a test that's not in the canonical-data should I note that?  I was thinking `total_after(1)` as a possible "edge" case and perhaps `total_after(3)` as a non-edge case since currently the only test for that function is an edge case with the Int128.

Comment (SaschaMann) [reply exercises/grains/runtests.jl]: Feel free to rename them.

Additional test cases don't need to be documented, though if they're potentially relevant to other languages  you can open a PR in the problem-spec repo and add them. Both tests sound reasonable to me.

**Comment by dhpollack** [comment]
@SaschaMann alright, should be good to go

**Comment by SaschaMann** [comment]
LGTM, thanks for improving the exercise!
